tensorflow_datasets/image/food101.py

```diff
@@ -69,26 +70,38 @@ def _info(self):
   def _split_generators(self, dl_manager):
     """Define Splits."""
 
-    path = dl_manager.download_and_extract(_BASE_URL)
+    dl_path = dl_manager.download_and_extract(_BASE_URL)
+    path = os.path.join(os.path.join(dl_path, "food-101"), "meta")
 
     return [
         tfds.core.SplitGenerator(
             name=tfds.Split.TRAIN,
             num_shards=4,
             gen_kwargs={
-                "data_dir_path": os.path.join(path, "food-101/images"),
+                "path": os.path.join(path, "train.json"),
+            },
+        ),
+
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            num_shards=1,
+            gen_kwargs={
+                "path": os.path.join(path, "test.json"),
             },
         ),
     ]
 
-  def _generate_examples(self, data_dir_path):
+  def _generate_examples(self, path):
     """Generate images and labels for splits."""
 
-    for class_name in tf.io.gfile.listdir(data_dir_path):
-      class_dir_path = os.path.join(data_dir_path, class_name)
-      for image_name in tf.io.gfile.listdir(class_dir_path):
-        image = os.path.join(class_dir_path, image_name)
-        yield image, {
-            "image": image,
-            "label": class_name,
-        }
+    data_dir_path = os.path.join(os.path.dirname(os.path.dirname(path)), 'images')
+    with tf.io.gfile.GFile(path) as f:
+        for line in f:
+            data = json.loads(line)
```

Comment:
Is this supposed to be like this? I mean, it looks like `data` will only contain the last line after all iterations are completed. And that last line will only be processed. Not sure how that's gonna work properly.

Maybe you meant this:
```python3
with tf.io.gfile.GFile(path) as f:
  data = json.loads(f.read())
. . . . .
```

Reply:
It doesn't make any difference as there is only a single line.  But, I would have to agree that the code given by you will take diff lines into consideration as well. I will update it.